### PR TITLE
[Python] Update docs for operators and streams

### PR DESCRIPTION
**Pull request by objorkman**

### CODE REVIEW

**Comment by objorkman** [comment]
I need to make some fixes to the other files, looks like the paths aren't working for the doctests

File: python/doc/source/operators.rst [hunk 1]
```diff
@@ -1,53 +1,61 @@
 Operators
 =========
 
-An ERDOS operator receives data on :py:class:`ReadStreams <erdos.ReadStream>`,
-and sends processed data on :py:class:`WriteStreams <erdos.WriteStream>`.
+ERDOS operators process received data, and use streams to broadcast Messages
```
Comment (pschafhalter) [inline python/doc/source/operators.rst:4]: Could you link to the `Message` and `WatermarkMessage` classes?

File: python/doc/source/operators.rst [hunk 2]
```diff
@@ -1,53 +1,61 @@
...
+implements a :py:class:`erdos.operator.Source` operator because it does not receive any data, 
+and sends messages on a single output stream.
+
+Operators can support both push and pull-based models of execution by 
+implementing methods defined for each operator. By implementing callbacks 
+such as :py:meth:`erdos.operator.OneInOneOut.on_data()`, operators can process 
+messages as they arrive. Moreover, operators can implement callbacks over watermarks 
+(e.g. :py:meth:`erdos.operator.OneInOneOut.on_watermark()`) to ensure ordered 
+processing over timestamps. ERDOS ensures lock-free, safe, and concurrent processing 
+by ordering callbacks in an ERDOS-managed execution lattice, which serves as a
```
Comment (pschafhalter) [inline python/doc/source/operators.rst:24]:
```suggestion
via a system-managed ordering of callbacks, which is implemented as a
```

File: python/doc/source/operators.rst [hunk 3]
```diff
@@ -1,53 +1,61 @@
...
-  :py:class:`ReadStreams <erdos.ReadStream>`.
-
-* The :py:meth:`~erdos.Operator.connect` method takes
-  :py:class:`ReadStreams <erdos.ReadStream>` and returns
-  :py:class:`WriteStreams <erdos.WriteStream>`
-  which are all later passed to :py:meth:`~erdos.Operator.__init__` by ERDOS.
-  The :py:class:`ReadStreams <erdos.ReadStream>` and
-  :py:class:`WriteStreams <erdos.WriteStream>`
-  must appear in the same order as in :py:meth:`~erdos.Operator.__init__`.
+Operators are structures which implement a certain communication pattern. For example,
```
Comment (pschafhalter) [inline python/doc/source/operators.rst:12]: Could you add that operators are implemented as classes, and must subclass the built-in operator type that specifies its communication pattern?

Also, describe the different types of communication patterns (Source, Sink, OneInOneOut, etc) in an unordered list

File: python/doc/source/streams.rst [hunk 4]
```diff
@@ -44,15 +42,10 @@ Operators use Write Streams to send data.
 Receiving Messages
 ------------------
 
-Operators receive data by registering callbacks or manually reading messages
-from Read Streams.
-
-Callbacks are functions which take an ERDOS message and any necessary write
-streams as arguments. Generally, callbacks process received messages and
-publish the results on write streams.
+Operators receive data by reading messages from Read Streams.
```
Comment (pschafhalter) [inline python/doc/source/streams.rst:45]: Could you link to the ReadStream class?

Comment (pschafhalter) [reply python/doc/source/streams.rst:45]: Operators also receive data by implementing callbacks that are automatically invoked upon the receipt of a message.

**Review by pschafhalter — CHANGES_REQUESTED** [review]
Could you make sure that the pythondoc tests pass? You might need to add `erdos.internal` to the mock modules listed here:
https://github.com/erdos-project/erdos/blob/3b1235a8274bfe84accef03f24da97c8caa679d6/python/doc/source/conf.py#L49

**Comment by pschafhalter** [comment]
Closing this PR and continuing work on #206

**Pull request closed without merging**